tests/core/test_data.py

```diff
@@ -0,0 +1,153 @@
+"""Unit tests for core.data module."""
+import os
+import shutil
+import tempfile
+import unittest
+
+import relentless
+
+class test_Directory(unittest.TestCase):
+    """Unit tests for core.data.Directory."""
+    def test_init(self):
+        """Test basic creation and methods of Directory."""
+        f = tempfile.TemporaryDirectory()
+
+        cwd = os.getcwd()
+
+        #test creation with existing path
+        d = relentless.data.Directory(f.name)
+        self.assertEqual(d.path, f.name)
+        self.assertEqual(d._start, None)
+        #enter and exit
+        with d:
+            self.assertEqual(d._start, cwd)
```

Comment:
When you are inside the context, you should also check that the current directory is right:
```py
self.assertEqual(os.getcwd(), f.name)
```